ios/OpenBot/OpenBot/Controllers/HomePage/HomePageViewController.swift

```diff
@@ -14,35 +14,70 @@ var viewControllerName: String?
+        label.textAlignment = .left
+        label.font = UIFont.boldSystemFont(ofSize: 20) // Change the font to bold and size to 20
+        self.addSubview(label)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
 
-class HomePageViewController: CameraController {
+class HomePageViewController: CameraController,UICollectionViewDataSource,UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
     @IBOutlet weak var bluetooth: UIButton!
     @IBOutlet weak var settings: UIButton!
     @IBOutlet weak var titleLabel: UILabel!
     @IBOutlet var modesCollectionView: UICollectionView!;
+    let sectionHeaders = ["  General", "  AI"]
 
     /// Called after the view controller has loaded.
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        UITabBar.appearance().tintColor = traitCollection.userInterfaceStyle == .dark ? UIColor.white: UIColor.black;
         bluetoothDataController.shared.startScan()
         DeviceCurrentOrientation.shared.findDeviceOrientation()
         setUpTitle();
         let layout = UICollectionViewFlowLayout();
         layout.collectionView?.layer.shadowColor = Colors.gridShadowColor?.cgColor
         layout.collectionView?.layer.shadowOpacity = 1
         if currentOrientation == .portrait {
-            layout.itemSize = resized(size: CGSize(width: width * 0.42, height: width * 0.42), basedOn: dimension)
+            layout.itemSize = resized(size: CGSize(width: width * 0.15, height: width * 0.15), basedOn: dimension)
         } else {
-            layout.itemSize = resized(size: CGSize(width: width * 0.42, height: width * 0.42), basedOn: dimension)
+            layout.itemSize = resized(size: CGSize(width: width * 0.15, height: width * 0.15), basedOn: dimension)
         }
-
+        layout.collectionView?.backgroundColor = Colors.voltageDividerColor
         layout.minimumInteritemSpacing = 5
         layout.sectionInset = UIEdgeInsets(top: 0, left: 8, bottom: 0, right: 8)
         layout.minimumLineSpacing = 30
-        modesCollectionView.collectionViewLayout = layout;
-        modesCollectionView.register(modesCollectionViewCell.nib(), forCellWithReuseIdentifier: modesCollectionViewCell.identifier)
+        layout.collectionView?.backgroundColor = Colors.voltageDividerColor
+        
         modesCollectionView.delegate = self
         modesCollectionView.dataSource = self
+        modesCollectionView.collectionViewLayout = layout;
+        modesCollectionView.register(modesCollectionViewCell.self, forCellWithReuseIdentifier: modesCollectionViewCell.identifier)
+        modesCollectionView.register(HeaderView.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: "Header")
+        self.view.addSubview(modesCollectionView)
+        
+        // Layout constraints
```

Comment:
Are these comments still needed / helpful for anything?

Reply:
Removed